services/repository/template.go

```diff
@@ -107,7 +107,8 @@ func GenerateRepository(ctx context.Context, doer, owner *user_model.User, templ
 	}()
 
 	// 2 - check whether the repository with the same storage exists
-	isExist, err := gitrepo.IsRepositoryExist(ctx, generateRepo)
+	var isExist bool
+	isExist, err = gitrepo.IsRepositoryExist(ctx, generateRepo)
```

Comment:
I do not think it is a bug

Reply:
Show me why

```

func TestError(t *testing.T) {
	f := func(t *testing.T) (err error) {
		defer func() {
			assert.ErrorContains(t, err, "error")
		}()
		val, err := 1, nil
		_ = val
		_ = err
		assert.NoError(t, err)
		return errors.New("error")
	}
	assert.ErrorContains(t, f(t), "error")
}
```

Reply:
OK. I see. The return will assign the error to the returned parameter err.